firestore-bigquery-export/guides/GENERATE_SCHEMA_VIEWS.md

```diff
@@ -0,0 +1,374 @@
+      <td>Type conversion (see previous section) will fail and the resulting column
+        will contain a BigQuery <code>null</code> value in lieu of the desired
+        value.</td>
+    </tr>
+    <tr>
+      <td>Specifying a schema field that doesn't exist in the underlying raw
+        changelog</td>
+      <td>Querying the column for that field will return a BigQuery <code>null</code>
+        value instead of the desired value.</td>
+    </tr>
+    <tr>
+      <td>Writing invalid JSON</td>
+      <td>The schema-views script cannot generate a view</td>
+    </tr>
+  </tbody>
+</table>
+
+Since all document data is stored in the schemaless changelog, mistakes in
+schema configuration don't affect the underlying data and can be resolved by
+re-running the schema-views script against an updated schema file.
+
+## About Schema Views
+
+### Views created by the script
+
++   `${param:PROJECT_ID}.${param:DATASET_ID}.${param:TABLE_ID}_schema_${SCHEMA_FILE_NAME}_changelog`
+
+    This view is a table which contains all rows present in the raw changelog.
+    This view is analogous to the raw change-log, only it has typed columns
+    corresponding to fields of the schema.
+
++   `${param:PROJECT_ID}.${param:DATASET_ID}.${param:TABLE_ID}_schema_${SCHEMA_FILE_NAME}_latest`  
+  
+    This view stores typed rows for the documents currently in the collection.
+    This view is analogous to the "latest" view on the raw changelog, only it
+    includes the typed columns corresponding to fields in the corresponding
+    schema file.
+
+    Since `GEOGRAPHY` fields are not groupable entities in BigQuery (and the
+    query which builds the latest view of a collection of documents requires
+    grouping on the schema columns), the latest schema omits any `GEOGRAPHY`
+    columns and, instead, splits them out into two `NUMERIC` columns called
+    `${FIRESTORE_GEOPOINT}_latitude` and `${FIRESTORE_GEOPOINT}_longitude`.
+
+### Columns in a schema view
+
+Each schema view carries with it the following fields from the raw changelog:
+
++   `document_name STRING REQUIRED`
++   `timestamp TIMESTAMP REQUIRED`
++   `operation STRING REQUIRED`
+
+The remaining columns correspond to fields of the schema and are assigned types
+based on the corresponding Cloud Firestore types those fields have. With the
+exception of `map` and `array`, the type conversion scheme is as follows:
+
+<table>
+  <thead>
+    <tr>
+      <th><strong>Cloud Firestore type</strong></th>
+      <th><strong>BigQuery type</strong></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>string</td>
+      <td>STRING</td>
+    </tr>
+    <tr>
+      <td>boolean</td>
+      <td>BOOLEAN</td>
+    </tr>
+    <tr>
+      <td>number</td>
+      <td>NUMERIC</td>
+    </tr>
+    <tr>
+      <td>timestamp</td>
+      <td>TIMESTAMP</td>
+    </tr>
+    <tr>
+      <td>geopoint</td>
+      <td>GEOGRAPHY</td>
+    </tr>
+    <tr>
+      <td>null</td>
+      <td>NULL</td>
```

Comment:
We should probably also include Firestore `reference` types here. Their corresponding values are "STRING"s containing the path to the referenced document.